src/microk8scluster.py

```diff
@@ -461,6 +461,8 @@ def _microk8s_kubeconfig(self, event):
         kubeconfig = yaml.safe_load(subprocess.check_output(["/snap/bin/microk8s", "config"]))
 
         public_address = subprocess.check_output(["unit-get", "public-address"]).decode().strip()
+        if ":" in public_address:
+            public_address = "[{}]".format(public_address)
```

Comment:
Did you consider using the `ipaddress` package from Python to properly check whether this is an IPv6 address instead? 

```py
In [9]: import ipaddress

In [10]: ipaddress.ip_address("192.168.0.1").version
Out[10]: 4

In [11]: ipaddress.ip_address('::1').version
Out[11]: 6

In [12]: ipaddress.ip_address('my.example.com').version
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-12-a41e44ca8ba6> in <module>
----> 1 ipaddress.ip_address('my.example.com').version

/usr/lib/python3.8/ipaddress.py in ip_address(address)
     51         pass
     52 
---> 53     raise ValueError('%r does not appear to be an IPv4 or IPv6 address' %
     54                      address)
     55 

ValueError: 'my.example.com' does not appear to be an IPv4 or IPv6 address
```

For example, this could look like this:

```py
try:
    if ipaddress.ip_address(public_address).version == 6:
        public_address = "[{}]".format(public_address)
except ValueError:
    pass
```

It's a little more verbose but feels safer. Any objections?

Reply:
Sure, question of personal taste I guess :)